lib/parser/source/comment/associator.rb

```diff
@@ -184,12 +184,19 @@ def associate_and_advance_comment(node)
         advance_comment
       end
 
+      MAGIC_COMMENT_RE = /^#\s*(-\*-|)\s*(frozen_string_literal|warn_indent|warn_past_scope):.*\1$/
+
       def advance_through_directives
         # Skip shebang.
         if @current_comment && @current_comment.text =~ /^#!/
           advance_comment
         end
 
+        # Skip magic comments.
```

Comment:
I thought you could put multiple directives in a single magic comment. Is that not so?

Reply:
Yeah you can. Parser does it https://github.com/whitequark/parser/blob/a31d2c0926fb084a3afe06feb0473e8ca5e398b7/Rakefile#L37

Here it is in the parse.y https://github.com/ruby/ruby/blob/3ba9e6b639afaf4c88145d56c9bd6746af16cfb7/parse.y#L7188-L7207

Reply:
My implementation ignores the entire comment line if it matches any magic comment, so that should be fine.